### PR TITLE
Add support for Application Emoji

### PR DESCRIPTION
**Pull request by Doc94**
**Description:** This PR add Support for Application Emoji (List, Get, Create, Update, Delete) also make a improvement for the GuildEmoji based in how Guild Emojis and Application Emoji share the same data (ignoring roles)

### CODE REVIEW

File: core/src/main/java/discord4j/core/object/entity/ApplicationEmoji.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        return Snowflake.of(applicationId);
+    }
+
+    /**
+     * Requests to retrieve the application this emoji is associated to.
+     *
+     * @return A {@link Mono} where, upon successful completion, emits the {@link ApplicationInfo application} this emoji is associated
+     * to. If an error is received, it is emitted through the {@code Mono}.
+     */
+    public Mono<ApplicationInfo> getApplication() {
```
Comment (Azn9) [inline core/src/main/java/discord4j/core/object/entity/ApplicationEmoji.java:68]: The emoji might not be associated with the current application if it is read from a message sent by an app. (I don't know how these would be parsed tho)

File: core/src/main/java/discord4j/core/object/entity/ApplicationEmoji.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+     *
+     * @param spec an immutable object that specifies how to edit this emoji
+     * @return A {@link Mono} where, upon successful completion, emits the edited {@link ApplicationEmoji}. If an error is
+     * received, it is emitted through the {@code Mono}.
+     */
+    public Mono<ApplicationEmoji> edit(ApplicationEmojiEditSpec spec) {
+        Objects.requireNonNull(spec);
+        return Mono.defer(
+                () -> this.getClient().getRestClient().getEmojiService()
+                        .modifyApplicationEmoji(getApplicationId().asLong(), getId().asLong(), spec.asRequest()))
```
Comment (Azn9) [inline core/src/main/java/discord4j/core/object/entity/ApplicationEmoji.java:94]:
```suggestion
                        .modifyApplicationEmoji(this.applicationId, getId().asLong(), spec.asRequest()))
```

File: core/src/main/java/discord4j/core/object/entity/ApplicationEmoji.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+    /**
+     * Requests to delete this emoji.
+     *
+     * @return A {@link Mono} where, upon successful completion, emits nothing; indicating the emoji has been deleted.
+     * If an error is received, it is emitted through the {@code Mono}.
+     */
+    public Mono<Void> delete() {
+        return this.getClient().getRestClient().getEmojiService()
+            .deleteApplicationEmoji(getApplicationId().asLong(), getId().asLong());
```
Comment (Azn9) [inline core/src/main/java/discord4j/core/object/entity/ApplicationEmoji.java:106]:
```suggestion
            .deleteApplicationEmoji(this.applicationId, getId().asLong());
```

File: core/src/main/java/discord4j/core/object/entity/ApplicationInfo.java [hunk 4]
```diff
@@ -461,6 +464,62 @@ public Flux<ApplicationRoleConnectionMetadata> setRoleConnectionMetadata(List<Ap
...
+    }
+
+    /**
+     * Requests to retrieve the guild emoji as represented by the supplied ID.
+     *
+     * @param id The ID of the guild emoji.
+     * @return A {@link Mono} where, upon successful completion, emits the {@link GuildEmoji} as represented by the
+     * supplied ID. If an error is received, it is emitted through the {@code Mono}.
+     */
+    public Mono<ApplicationEmoji> getApplicationEmojiById(final Snowflake id) {
```
Comment (Azn9) [inline core/src/main/java/discord4j/core/object/entity/ApplicationInfo.java:488]: Either rename this method "getEmojiById" or just "getEmoji" to keep a similarity between all methods names.

**Review by Azn9 — CHANGES_REQUESTED** [review]

File: core/src/main/java/discord4j/core/object/entity/ApplicationInfo.java (hunk 4 above)
Comment (Doc94) [reply core/src/main/java/discord4j/core/object/entity/ApplicationInfo.java:488]: i based in Guild where has the same names. https://github.com/Discord4J/Discord4J/blob/3.2.x/core/src/main/java/discord4j/core/object/entity/Guild.java#L533

File: core/src/main/java/discord4j/core/object/entity/ApplicationEmoji.java (hunk 1 above)
Comment (Azn9) [reply core/src/main/java/discord4j/core/object/entity/ApplicationEmoji.java:68]: Oh, okay then 👌

**Review by Azn9 — APPROVED** [review]

**Pull request merged**